st2common/st2common/util/schema/__init__.py

```diff
@@ -136,7 +137,9 @@ def assign_default_values(instance, schema):
         schema_items = property_data.get('items', {})
 
         # Array
-        if attribute_type == 'array' and schema_items and schema_items.get('properties', {}):
+        if (schema_items and schema_items.get('properties', {}) and
+                (attribute_type == 'array' or
+                    (isinstance(attribute_type, list) and 'array' in attribute_type))):
```

Comment:
The list needs to be supported for the inline schema change to allow None. For optional parameters where None is allowed, internally we create a list and put the original param type in a list along with null (i.e. ['string', 'null'], ['number', 'null'], ['array', 'null']).  This is the same with an param with type array or object. Since array and object are processed separately, we need to account for use cases where these params are optional and none is allowed.